lib/configure.js

```diff
@@ -312,6 +322,42 @@ function configure (gyp, argv, callback) {
 }
 
 /**
+ * Returns value of NODE_MODULE_VERSION define
+ * from Node source code in specified directory
+ * for specified Node version
+ */
+function nodeModulesVersion(nodeDir) {
+  var versionStr = path.basename(nodeDir)
+
+  // parse version from the base name of nodeDir
+  // probably removing optional `releaseName-` prefix
+  var nodeVersion = semver.valid(versionStr)
+                 || semver.valid(versionStr.substr(versionStr.indexOf('-') + 1))
+
+  if (!nodeVersion) {
+    throw new Error('Can\'t extract target version from ' + nodeDir)
+  }
+
+  // NODE_MODULE_VERSION may be defined in different files
+  // depending on the target Node.js version
+  if (semver.gte(nodeVersion, '3.0.0')) {
+    nodeVersion = 'include/node/node_version.h'
+  } else if (semver.gt(nodeVersion, '0.11.4')) {
+    nodeVersion = 'src/node_version.h'
+  } else {
+    nodeVersion = 'src/node.h'
+  }
+  nodeVersion = path.join(nodeDir, nodeVersion)
+
+  var modulesVersion = fs.readFileSync(nodeVersion, { encoding: 'utf8'})
```

Comment:
Style: space before `}`.

You should wrap the `fs.readFileSync` call in a try/catch in case the header file is missing.